### DISCUSSION TITLE
Change performance overlay

### DISCUSSION THREAD

**Discussion by alessioferri20** [Valkirie/HandheldCompanion#743] [Q&A] [answered]
Is there any way to change the performance overlay available?

**Comment by CasperH2O** [comment]
The overlays are currently hardcoded so to speak. You can add or edit them here: https://github.com/Valkirie/HandheldCompanion/blob/main/HandheldCompanion/Managers/OSDManager.cs

An example of an addition can be found in this PR: https://github.com/Valkirie/HandheldCompanion/pull/648

**Reply by alessioferri20** [reply]
After edit and recompiled how to test it? And after testing which files/dll I need to update in the installation folder for everything to work?

**Reply by alessioferri20** [reply]
After some testing I have done it and it seems to work; I have added frametime to the full overlay, if you are interested I can do a pull request!

**Reply by CasperH2O** [reply]
Can you share a screenshot?

**Reply by alessioferri20** [reply]
Sure, no problem!
![Immagine 2023-07-27 092816](https://github.com/Valkirie/HandheldCompanion/assets/23658291/830ca2cb-e92d-4f94-9365-09c0ea232387)

**Reply by alessioferri20** [reply]
Here is the code if you need it:
- Here I have added the entry after the FPS  in OSDManager.cs using a new method in RTSS.cs
![Immagine 2](https://github.com/Valkirie/HandheldCompanion/assets/23658291/7dc89cce-6f46-4afd-ad9f-b74100abc4f9)

- Here is the new method in RTSS.cs
![Immagine 3](https://github.com/Valkirie/HandheldCompanion/assets/23658291/a94927ba-9742-4ac3-bc28-32282ed8cd97)

- Here I have made a slight modification to the ToString method used to display every property in order to achieve the double format for the frametime in OSDManager.cs
![Immagine 4](https://github.com/Valkirie/HandheldCompanion/assets/23658291/d14f9cff-a3e2-414c-80e9-3ef76ee0ce79)

- Here is the new ToString method with the double type format in HWiNFO.cs
![Immagine 5](https://github.com/Valkirie/HandheldCompanion/assets/23658291/76dc5450-eae7-420d-af8b-d8b1f5c440d4)

**Comment by CasperH2O** [comment]
@Valkirie what do you think, shall we ask @alessioferri20 to do a PR? Looks good to me.

**Reply by alessioferri20** [reply]
In case he doesn't answer or like the changes which files should I change in the Program Folder? Just the exe? Or some other dlls?

**Reply by alessioferri20** [reply]
@CasperH2O ?

**Reply by CasperH2O** [reply]
@alessioferri20 I honestly don't know. You could make a change in the code, compile and check the latest changed files?

**Reply by alessioferri20** [reply]
@CasperH2O Should I push that pull request?

**Reply by CasperH2O** [reply]
@alessioferri20 I think yes, but final decision is up to @Valkirie and he is currently on holiday.

**Comment by CasperH2O** [comment, marked as answer]
Closing this as there is a design and some prototyping ongoing in #744
